First pass of a v4 cookbook

glenrobson opened the pull request:
Steps for adding a v4 recipe for example `0001-mvm-image`:

1. Make a directory in `0001-mvm-image` called `v4`.
2. Rename the index.md in `0001-mvm-image` to recipe.md
3. copy everything in `0001-mvm-image` to v4 
4. Edit the manifest.json and recipe.md in `v4` to make it work for v4
5. Add a index.json similar to the one in this pull requests which will show both recipes. 

Notes:
 * Frontmatter is only present in the index.md parent recipe so all topics, viewers and other things need to go in this file rather than the child recipes
 * added a new field to index.md `v4-viewers` to list v4 supporting recipes 
 * Need a more elegant way of linking to v3 only recipes in a v4 recipe. 
 * Maybe highlight which versions of a recipe are aviliable when linking to a recipe. 
 * Will need to think about validation of v4 and sending v3 files to v3 validator and vise versa.

glenrobson committed:
First pass of a v4 recipe

glenrobson committed:
Pausing validation

glenrobson committed:
Reversing if statement to default to 3

glenrobson committed:
Don't need to specify supported versions

glenrobson commented:
Example v3 and v4 recipe: https://preview.iiif.io/cookbook/v4/recipe/0001-mvm-image/

glenrobson committed:
Adding a v3/v4 link to homepage

glenrobson committed:
Making version more prominent and defaulting to 3

glenrobson committed:
Fixing link

glenrobson commented on recipe/0001-mvm-image/v4/recipe.md:
```diff
@@ -0,0 +1,33 @@
+
+This illustrates the mandatory structure and properties of a manifest, with the simplest possible content. 
+
+The JSON-LD opens with the `@context` declaration, which identifies the terms used in the document as belonging to the IIIF specification. The `id` property identifies this manifest with the URL at which it is available online. The `type` property must be `Manifest`. The `label` property is mandatory, and the language of its value must be given (or the special value `none`), using a [language map ...
```
Point to version 4 of the spec: [language map][prezi3-languages]

triplingual committed:
Updated for IIIF Presentation v4 🆙

triplingual committed:
Added new files for v4 structure and content ➕

Initial commit to version control ⚡️

triplingual committed:
Increased clarity and specificity of label value 🛠️

triplingual committed:
Added highlighting for new-to-v4 Type value 🎨

triplingual committed:
Added convenience links to the two parts of the v4 spec ➕

triplingual committed:
Mentioned v4 difference and linked to appropriate spec section 📝

triplingual committed:
Updated Implementation Notes body text for v4 📝

triplingual committed:
Removed incorrect and unnecessary info 🚮

triplingual committed:
Re-centered recipe title 🎨

triplingual committed:
Edited per cookbook authors call to leave behind comparisons to MVM Image 📝👋🏻

triplingual committed:
Reworded related recipe text, added 3D related 📝⛓️

triplingual committed:
Moved highlighting to the correct template place 🚚🤦🏻‍♂️

triplingual committed:
Added entry for prezi4 version of the recipe ➕

triplingual committed:
Sorted links numerically for readability 🔢

robcast committed:
first version of updated simplest video recipe for v4

triplingual committed:
Removed duplicative TOC entry 🚮

triplingual committed:
Updated IDs semantically and functionally (in the case of target) 📝🆙

robcast committed:
add v4 magic and links.

robcast committed:
fix more v4 path issues.

glenrobson committed:
Merge pull request #611 from IIIF/v4-0003-simplest-manifest-video

add V4 version of 0003 simplest manifest video

triplingual committed:
Merge branch 'v4' into v4-0002-simplest-manifest-audio

triplingual committed:
Merge pull request #607 from IIIF/v4-0002-simplest-manifest-audio

0002 Minimum viable audio v3/v4

glenrobson committed:
Fixing links to keep in v4 area of cookbook

glenrobson committed:
Adding 3d model

glenrobson committed:
Removing viewers

glenrobson committed:
Merge pull request #612 from IIIF/v4-fixes-3d

V4 fixes 3d

glenrobson committed:
Update links.md

glenrobson committed:
Merge pull request #619 from IIIF/glenrobson-patch-1

Update links.md

stephenwf commented:
@glenrobson I've temporarily fixed the Audio/Video Presentation 4 support in Theseus. (Until the parser has full support for the 3D additions)

triplingual committed:
Added version parenthetical to `label` 📝

triplingual committed:
Added version parenthetical to `label` 📝

triplingual committed:
Conformed manifest `label` text to other foundational recipes

triplingual committed:
Copyedited ✍🏻

triplingual committed:
Edited significantly 📝

For parallelism to other foundational recipes, for completeness, and for readability

triplingual committed:
Edited further for readability and simplicity 📝

triplingual committed:
Conformed `label` to other Simplest manifests

triplingual committed:
Merge branch 'master' into v4

triplingual committed:
Created 2 alternative presentation approaches for recipes with both 3 and 4 ✨

triplingual committed:
Removed 2.1 fixtures from created recipes 🚮

triplingual committed:
Removed issue-only / issue-less recipes 🚮

Ensured all removed entries had issues created for them

triplingual committed:
Added head text about versions ➕📝

triplingual committed:
Removed in-process recipes 🚮

triplingual committed:
Rephrased comment in Other Annotations section for a public audience 📝

triplingual committed:
Edited comment 📝

I'm surely overthinking this, but I don't want readers to skip these annotations as if they are _only_ building blocks

triplingual committed:
Converted `target`s into arrays ♻️

triplingual committed:
Removed duplicative entry, added language map link 📝

triplingual committed:
Swapped GitHub location for IIIF Fixtures one 🔂

triplingual committed:
Significant editing (see additional comments) 📝

These edits both bring the 4 minimum viable manifest recipes more in line with each other and also highlight differences better. With v4, there are more significant differences, so for instance the audio recipe needs to spell out more what it shares with the other kind of manifests.

triplingual commented on recipe/0608-mvm-3d/v4/manifest.json:
```diff
@@ -0,0 +1,33 @@
+              "id": "{{ id.path }}/scene/1/annotationPage/1/anno/1",
+              "type": "Annotation",
+              "motivation": ["painting"],
+              "body": {
+                "id": "https://raw.githubusercontent.com/IIIF/3d/main/assets/astronaut/astronaut.glb",
+                "type": "Model",
+                "format": "model/gltf-binary"
+              },
```
The current API RC describes `body` as being a (singular) JSON object, if I'm reading correctly.

triplingual committed:
Fixed target's target, h/t @stephenwf 🛠️

triplingual commented on recipe/0002-mvm-audio/v4/manifest.json:
```diff
@@ -0,0 +1,33 @@
+                "format": "audio/mp4",
+                "duration": 1985.024
+              },
+              "target": "{{ id.path }}/timeline"
```
As with `body`, the current RC has target as a singular JSON object. [https://iiif.io/api/presentation/4.0/model/#target](https://iiif.io/api/presentation/4.0/model/#target)

triplingual commented on recipe/0003-mvm-video/v4/manifest.json:
```diff
@@ -0,0 +1,37 @@
+                    "duration": 572.034,
+                    "format": "video/mp4"
+                },
+                "target": "{{ id.path }}/canvas"
```
Current RC has `target` as a singular JSON object. [https://iiif.io/api/presentation/4.0/model/#target](https://iiif.io/api/presentation/4.0/model/#target)

triplingual commented on recipe/0608-mvm-3d/v4/manifest.json:
```diff
@@ -0,0 +1,33 @@
+                "type": "Model",
+                "format": "model/gltf-binary"
+              },
+              "target": "{{ id.path }}/scene/1/annotationPage/1"
```
Fixed the referenced part, but left the syntax as current RC says "a JSON object"

triplingual committed:
Merge branch 'master' into v4

glenrobson committed:
Merge branch 'master' into v4

glenrobson committed:
Fixing validation error

glenrobson committed:
Fixing frontmatter

glenrobson committed:
Adding v4 matrix

glenrobson committed:
Fix styling on v3 only recipes

glenrobson committed:
Moving links to recipes to v3

glenrobson committed:
Adding a JSON diff to v3 + v4 recipes

glenrobson committed:
Merge branch 'master' into v4

glenrobson committed:
Renaming diff

glenrobson committed:
Try 1 at disabling auto-scroll to heading

glenrobson committed:
A more elegant way of stopping scrolling

glenrobson committed:
Updating viewer support

glenrobson committed:
Fixing v4 matrix

glenrobson committed:
Merge branch 'master' into v4

glenrobson committed:
Replacing with master version now that the validation is seperate

glenrobson committed:
Correcting link to the 3d recipe

glenrobson committed:
Correcting link

glenrobson committed:
Moving link to main spec page rather than model

glenrobson committed:
Removing empty viewer

glenrobson committed:
Introducing v3-code for code implementations

glenrobson committed:
Updating README with new variables

glenrobson committed:
Merge branch 'master' into v4

glenrobson committed:
Changing ids to be the same as v3 for image recipe

glenrobson committed:
Merge remote-tracking branch 'refs/remotes/origin/v4' into v4

glenrobson committed:
Fixing styling

glenrobson committed:
Adding note about Sound and updating use case

glenrobson committed:
Reusing ids

glenrobson committed:
Addressing Editor comments

glenrobson committed:
Addressing editor comments

glenrobson committed:
Adding 3d Viewers to the recipe

glenrobson committed:
Merge branch 'master' into v4
